filecoin-proofs/src/api/mod.rs

```diff
@@ -381,14 +381,13 @@ pub fn verify_piece_inclusion_proof(
 
 /// Takes a piece file at `unpadded_piece_path` and the size of the piece and returns the comm_p.
 ///
-pub fn generate_piece_commitment<T: Into<PathBuf> + AsRef<Path>>(
-    unpadded_piece_path: T,
+pub fn generate_piece_commitment(
+    unpadded_piece_file: impl std::io::Read,
```

Comment:
SUGGESTION: Use `T: Read` instead of `impl Read`

Reply:
fixed